Document line symbol layers properties (#4624)
qgis/QGIS-Documentation: source/docs/user_manual/style_library/symbol_selector.rst

* **Simple line** (default): available settings are:

  .. _figure_simple_line_symbol:

  .. figure:: img/simpleLineSymbol.png
     :align: center

     Designing a Simple Line Symbol

>   Along with common properties of the :ref:`simple marker symbol

Author: Actually, a line does not show ALL the properties of a marker, just some of them. And here I wanted a way to introduce only the new ones.

  the line feature must have at least three vertices. It also uses a
  :ref:`fill symbol <vector_fill_symbols>` such as gradients or shapeburst
  to render the arrow body. Combined with the geometry generator, this type of
  layer symbol helps you representing flow maps.
* **Geometry generator** (see :ref:`geometry_generator_symbol`)

.. _marker_line_symbol:

> * **Marker line**: allows for a repeating :ref:`marker symbol

Reviewer: ```suggestion
* **Marker line**: repeats a :ref:`marker symbol
```

>   <vector_marker_symbols>` to be drawn over the length of a line symbol.

Reviewer: ```suggestion
  <vector_marker_symbols>` over the length of a line.
```

>   * The markers placement can be at a regular distance or based on the line

Reviewer: ```suggestion
  * The markers can be placed with regular distances, or be based on the line
```

Author: It's the marker placement that is based on the line geometry, not the marker itself

Author: Not convinced this is better...

    geometry: first, last or each vertex, on the central point of the line
    or of each segment, or on every curve point.

>   * The markers placement can also be offset along the line

Reviewer: ```suggestion
  * The markers placement can also be given an offset along the line
```

  * The |checkbox| :guilabel:`Rotate marker` option allows you to set whether

>     each marker symbol should follow the line orientation or its own rotation.

Reviewer: ```suggestion
    each marker symbol should be oriented relative to the line direction or not.
```

Reviewer: This is important.  The specified "rotation" for the marker symbol will be applied also if "rotate marker" is checked.

Author: indeed

    Because a line is often a succession of segments of different directions,

>     the marker's rotation is calculated by averaging the line over a specified

Reviewer: ```suggestion
    the rotation of the marker is calculated by averaging over a specified
```

>     distance either side of the symbol. For example, setting the

Reviewer: ```suggestion
    distance along the line on. For example, setting the
```

>     :guilabel:`Average angle over` property to ``4mm`` means to take the points

Reviewer: ```suggestion
    :guilabel:`Average angle over` property to ``4mm`` means that the two points
```

>     along the line ``2mm`` from either side of the symbol placement, and use

Reviewer: ```suggestion
    along the line that are ``2mm`` before and after the symbol placement are used
```

>     these instead to calculate the line angle for that symbol.

Reviewer: ```suggestion
    to calculate the line angle for that marker symbol.
```

    This has the effect of smoothing (or removing) any tiny local deviations

>     from the overall line direction, resulting in much nicer visual orientation

Reviewer: ```suggestion
    from the overall line direction, resulting in much nicer visual orientations
```

>     of the marker line.

Reviewer: ```suggestion
    of the marker line symbols.
```

  * The marker line can also be offset from the line itself.

.. _hashed_line_symbol:

> * **Hashed line**: allows for a repeating line segment (a hash) to be drawn

Author: ```suggestion
* **Hashed line**: repeats a line segment (a hash)
```

  over the length of a line symbol, with a line sub-symbol used to render each
  individual segment. In other words, a hashed line is like a marker line in
  which marker symbols are replaced with segments. As such, the hashed lines

>   have :ref:`same properties <marker_line_symbol>` as marker line symbols,

Reviewer: ```suggestion
  have the :ref:`same properties <marker_line_symbol>` as marker line symbols,
```
